src/main/java/org/testng/xml/internal/TestNamesMatcher.java

```diff
@@ -0,0 +1,106 @@
+package org.testng.xml.internal;
+
+import java.util.Iterator;
+import java.util.List;
+import org.testng.TestNGException;
+import org.testng.collections.Lists;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+
+/**
+ * The class to work with "-testnames"
+ *
+ */
+public final class TestNamesMatcher {
+    
+    private final List<XmlSuite> cloneSuites = Lists.newArrayList();
+    private final List<String> matchedTestNames = Lists.newArrayList();
+    private final List<XmlTest> matchedTests = Lists.newArrayList();
+    private List<String> testNames = Lists.newArrayList();
+    
+    /**
+     * Constructor of this class. 
+     */
+    public TestNamesMatcher(XmlSuite xmlSuite, List<String> testNames){
+        this.testNames = testNames;
+        cloneIfContainsTestsWithNamesMatchingAny(xmlSuite, this.testNames);        
+    }
+
+    /**
+     * Recursive search the given testNames from the current {@link XmlSuite} and its child suites.
+     *
+     * @param xmlSuite  The {@link XmlSuite} to work with.
+     * @param testNames The list of testnames to iterate through
+     */
+    private void cloneIfContainsTestsWithNamesMatchingAny(XmlSuite xmlSuite, List<String> testNames) {
+        if (testNames == null || testNames.isEmpty()) {
+            throw new TestNGException("Please provide a valid list of names to check.");
+        }
+        
+        //Start searching in the current suite.
+        addIfNotNull(cloneIfSuiteContainTestsWithNamesMatchingAny(xmlSuite));
+        
+        //Search through all the child suites.
+        for (XmlSuite suite : xmlSuite.getChildSuites()) {
+            cloneIfContainsTestsWithNamesMatchingAny(suite, testNames);
+        }
+    }
+    
+    public List<XmlSuite> getCloneSuite() {
+        return cloneSuites;
+    }
+
+    /**
```

Comment:
Please help fix this java docs since it doesn't match with the method signature.

Reply:
Done.